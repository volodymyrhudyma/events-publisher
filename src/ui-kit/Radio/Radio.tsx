import React, { PureComponent, ReactNode } from 'react';
import Label from 'src/ui-kit/Label';
import { StyledRadio, Wrapper } from './styled/Radio.styled';

interface IRadioProps {
  value: string;
  label: string | ReactNode;
  checked: boolean;
  /* tslint:disable-next-line */
  onChange: (value: any) => void;
}

class Radio extends PureComponent<IRadioProps> {
  public render() {
    const { value, label, checked, onChange } = this.props;
    return (
      <Wrapper>
        <StyledRadio
          type="radio"
          value={value}
          checked={checked}
          onChange={onChange}
        />
        <Label
          uppercase={false}
          color="secondary"
          onClick={onChange.bind(this, value)}
        >
          {label}
        </Label>
      </Wrapper>
    );
  }
}

export default Radio;
