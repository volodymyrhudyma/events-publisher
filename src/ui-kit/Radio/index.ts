import Component from './Radio';

export default Component;
