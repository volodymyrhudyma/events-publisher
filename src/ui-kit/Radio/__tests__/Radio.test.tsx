import React from 'react';
import { shallow } from 'enzyme';
import Radio from '../Radio';

describe('Radio component', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Radio value="Value" checked={false} label="Label" onChange={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
