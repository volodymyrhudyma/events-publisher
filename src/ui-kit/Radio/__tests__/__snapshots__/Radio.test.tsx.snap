// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component should render correctly 1`] = `
<styled.div>
  <styled.input
    checked={false}
    onChange={[MockFunction]}
    type="radio"
    value="Value"
  />
  <Label
    color="secondary"
    onClick={[Function]}
    uppercase={false}
  >
    Label
  </Label>
</styled.div>
`;
